### ISSUE TITLE
GraphQL filtering by `_type` does not work

### ISSUE THREAD

**Issue by joewesch** [nautobot/nautobot#1589]
### Environment
* Python version: 3.8
* Nautobot version: 1.2.10

### Steps to Reproduce
Try and filter a device's interfaces to only virtual type using the GraphQL interface at [demo.nautobot.com](https://demo.nautobot.com/graphql/#query=%7B%0A%20%20devices(name%3A%20%22ams01-edge-01%22)%20%7B%0A%20%20%20%20name%0A%20%20%20%20interfaces(_type%3A%20%22virtual%22)%20%7B%0A%20%20%20%20%20%20name%0A%20%20%20%20%20%20type%0A%20%20%20%20%7D%0A%20%20%7D%0A%7D) with the following query:
```
{
  devices(name: "ams01-edge-01") {
    name
    interfaces(_type: "virtual") {
      name
      type
    }
  }
}
```

### Expected Behavior
The only interface that should be returned is `Loopback0`.

### Observed Behavior
All of the interfaces are returned.

The exclude filter, however, does seem to work:
```
{
  devices(name: "ams01-edge-01") {
    name
    interfaces(type__n: "virtual") {
      name
      type
    }
  }
}
```

**Comment by jathanism** [comment]
I confirmed this behavior. The root cause are two things:

1. We have an explicit chunk of code that is replacing any filter named `type` with `_type` because it clashes with the internal `Field.type` value for Graphene fields (https://github.com/nautobot/nautobot/blob/e999d8a13fe94bd1c0b5aebc16b331db3dba6436/nautobot/core/graphql/utils.py#L65-L73)
2. This causes the filterset to ignore incoming `_type=` filter arguments because they are not defined on the filterset. 

In hindsight this was a naive change. We'll need to come up with something. Either mapping an incoming query field to an underlying filterset field, or upgrading from Graphene 2.x to 3.x where the `Field` API has changed and this is no longer an issue.

**Comment by jathanism** [comment]
Fixed in https://github.com/graphql-python/graphene/pull/1216 and available in `graphene==3.0.0`.

**Comment by glennmatthews** [comment]
Fixed in `next` (1.4.0) by #1794.
